Library/Formula/aws-sdk-cpp.rb

```diff
@@ -0,0 +1,70 @@
+class AwsSdkCpp < Formula
+  desc "AWS SDK for C++"
+  homepage "https://github.com/aws/aws-sdk-cpp"
+  url "https://github.com/aws/aws-sdk-cpp/archive/0.9.6.tar.gz"
+  sha256 "cf2dad62e50c5c5cff88a4ad959af3aa402aa7e8aa1692bdf807052c0fe4aeb9"
+  head "https://github.com/aws/aws-sdk-cpp.git"
+
+  ALL_CLIENTS = [
```

Comment:
I'd love to find a solution that's agreeable to everyone...

What if we split it up into a flag essentially equivalent to each section in the AWS console?

`--with-compute`=ec2,...

I put together everything into the categories as well as I could based on the console and further research:
```
compute: ec2;elasticbeanstalk;lambda
storage: s3;cloudfront;elasticfilesystem;glacier;importexport;storagegateway
database: rds;dynamobdb;elasticache;redshift;dms;sdb
networking: directconnect;route53
devtools: codecommit;codedeploy;codepipeline
management: cloudformation;cloudtrail;config;opsworks;ecr;ecs;autoscaling;support;logs;monitoring
security: identity-management;access-management;inspector;waf;iam;cloudhsm;acm;ds;kms;sts
analytics: datapipeline;kinesis;firehose;marketplacecommerceanalytics
iot: iot
game: gamelift
mobile: cognito-identity;cognito-sync;devicefarm;mobileanalytics;sns
appservices: apigateway;cloudsearch;elastictranscoder;sqs;swf;email
enterprise: workspaces;

other: elasticloadbalancing;elasticmapreduce;es;events;machinelearning;queues;transfer
```

We could merge `iot`, `game` and `enterprise` in with `other` and/or eliminate `other` entirely to end up with 10ish flags that give a range of flexible options. Multiple options could be specified, and if none were specified, everything could be built.

Thoughts?

Below is a screenshot of the categories from the console:

<img width="989" alt="screen shot 2016-03-24 at 4 00 03 pm" src="https://cloud.githubusercontent.com/assets/575602/14034235/72527004-f1de-11e5-98c2-9afa2355bf62.png">